### ISSUE TITLE
Aggregated room directory search

### ISSUE THREAD

**Issue by notramo** [vector-im/element-web#15560]
**Is your suggestion related to a problem? Please describe.**
I configured multiple servers in my room explorer.

**Describe the solution you'd like**
If I want to search for a given term in the Matrix federation, I have to select each server one by one, enter the search term, and see if it has results.

**Describe alternatives you've considered**
Element should query all the configured servers upon entering the search term, and aggregate the results.

**Additional context**
![image](https://user-images.githubusercontent.com/26059643/97201331-3a9a4500-17aa-11eb-92a8-2c7faf88275c.png)

**Comment by t3chguy** [comment]
> Element should query all the configured servers upon entering the search term, and aggregate the results.

No, given that Element has to query via its homeserver anyway, the homeserver should do it.
Plus there's the difficult issue of trust.
What if Server A says #alias1 is !roomA but Server B says #alias1 is !roomB

**Comment by notramo** [comment]
> No, given that Element has to query via its homeserver anyway, the homeserver should do it.
But it needs an UI to indicate what server has the given room in the directory, so it is not necessary to change server. Element can query all the servers via the HS, but it should do the aggregation locally (especially given the following problem).

> What if Server A says #alias1 is !roomA but Server B says #alias1 is !roomB

In this case, this should be indicated with red warnings.
Without the aggregated solution, this wouldn't be even noticed. (The user joins the room when first found on any of the servers.)

**Comment by t3chguy** [comment]
The aggregation should be done by the server, would be much cheaper to do 1 API hit rather than N where N can be infinite.
